testing/dub: new aport

Geod24 opened the pull request:
Add dub v1.18.0, the D package manager.

Geod24 commented:
CC @Cogitri

maxice8 commented on testing/dub/APKBUILD:
```diff
@@ -0,0 +1,29 @@
+builddir="$srcdir/dub-$pkgver/"
+
+build() {
+	DMD=ldmd2 GITVER="v${pkgver}" ./build.sh
```
unnecessary usage of brackets ${ }

maxice8 commented on testing/dub/APKBUILD:
```diff
@@ -0,0 +1,29 @@
+depends="libcurl"
+makedepends="ldc ldc-static bash curl-dev"
+subpackages="$pkgname-doc"
+source="https://github.com/dlang/dub/archive/v${pkgver}.zip"
```
unnecessary usage of brackets ${ }

maxice8 replied:
if possible use the .tar.gz, it is **highly** preferred over .zip files.

maxice8 replied:
prefix the URL with `$pkgname-$pkgver.tar.gz::`

maxice8 commented on testing/dub/APKBUILD:
```diff
@@ -0,0 +1,29 @@
+	DMD=ldmd2 GITVER="v${pkgver}" ./build.sh
+}
+
+check () {
```
remove the space between `check` and `()`

maxice8 commented on testing/dub/APKBUILD:
```diff
@@ -0,0 +1,29 @@
+}
+
+check () {
+    bin/dub test
```
indent with tabs, not spaces

maxice8 commented on testing/dub/APKBUILD:
```diff
@@ -0,0 +1,29 @@
+makedepends="ldc ldc-static bash curl-dev"
+subpackages="$pkgname-doc"
+source="https://github.com/dlang/dub/archive/v${pkgver}.zip"
+builddir="$srcdir/dub-$pkgver/"
```
This can be removed as it is the default value

maxice8 commented on testing/dub/APKBUILD:
```diff
@@ -0,0 +1,29 @@
+}
+
+package() {
+	install -D "$builddir/LICENSE" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
```
It doesn't necessarily need to be installed but it is your call

Geod24 commented:
Addressed all of @maxice8 review's except for the `zip` one. When I switched from `zip` to `tar.gz`, I got an unpacking error:
```console
>>> dub: Unpacking /var/cache/distfiles/dub-1.18.0.tar.gz...
tar: dub-1.18.0/test/issue1336-registry/packages/gitcompatibledubpackage/1.0.2.zip: Cannot utime: No such file or directory
tar: Exiting with failure status due to previous errors
>>> ERROR: dub: unpack failed
```

The file in question is a symbolic link. Is there any issue related to symlink with Alpine's `tar` ?

maxice8 commented on testing/dub/APKBUILD:
```diff
@@ -0,0 +1,26 @@
+url="http://code.dlang.org/"
+arch="x86_64"
+license="MIT"
+depends="libcurl"
```
is this really required ? from my testing it doesn't link to  libcurl automatically so no binary uses it

Geod24 replied:
Yes. Phobos (the D standard library) does not link to `curl`, instead it loads it [at runtime](https://github.com/dlang/phobos/blob/e70d86b75d82f756ceec3b1398b77d1c1e9248bd/std/net/curl.d#L4165-L4195)

wilzbach replied:
It's possible to build `dub` without curl though (if desired) by relying on Vibe.d (see the `net` config). However, this requires one to have a dub build in the first place, so this could only be done after bootstrapping.

maxice8 commented on testing/dub/APKBUILD:
```diff
@@ -0,0 +1,26 @@
+license="MIT"
+depends="libcurl"
+makedepends="ldc ldc-static bash curl-dev"
+source="$pkgname-$pkgver.zip::https://github.com/dlang/dub/archive/v$pkgver.zip"
```
There is a .tar.gz tarball autogenerated by GitHub like the .zip, isn't it usable ?
